subxt/src/backend/rpc/rpc_client_t.rs

```diff
@@ -101,3 +101,94 @@ impl<T: RpcClientT> RpcClientT for Box<T> {
+        params: Option<Box<RawValue>>,
+        unsub: &'a str,
+    ) -> RawRpcFuture<'a, RawRpcSubscription> {
+        use futures::{FutureExt, StreamExt, TryStreamExt};
+        use reconnecting_jsonrpsee_ws_client::{RpcError as ReconnError, SubscriptionId};
+
+        // Helper for more human friendly error messages.
+        // Workaround for: https://github.com/paritytech/jsonrpsee/issues/1280
+        fn rpc_error(err: ReconnError) -> String {
+            match err {
+                ReconnError::Transport(e) => e.to_string(),
+                e @ ReconnError::InvalidSubscriptionId => e.to_string(),
+                ReconnError::InvalidRequestId(e) => e.to_string(),
+                ReconnError::Custom(e) => e,
+                ReconnError::RestartNeeded(inner_err) => {
+                    // HACK: jsonrpsee has Error::RestartNeeded(Arc<Error>) and
+                    // we need to visit one child because
+                    // RestartNeeded only contains one layer of recursion.
+                    //
+                    // Thus, `RestartNeeded(RestartNeededArc<Error>)`
+                    // and is unreachable.
+                    match &*inner_err {
+                        ReconnError::Transport(e) => e.to_string(),
+                        e @ ReconnError::InvalidSubscriptionId => e.to_string(),
+                        ReconnError::InvalidRequestId(e) => e.to_string(),
+                        ReconnError::Custom(e) => e.to_string(),
+                        // These variants are infallible.
+                        ReconnError::RestartNeeded(_)
+                        | ReconnError::RequestTimeout
+                        | ReconnError::MaxSlotsExceeded
+                        | ReconnError::Call(_)
+                        | ReconnError::ParseError(_)
+                        | ReconnError::HttpNotImplemented
+                        | ReconnError::EmptyBatchRequest(_)
+                        | ReconnError::RegisterMethod(_) => unreachable!(),
+                    }
+                }
+                // These variants are infallible.
+                ReconnError::RequestTimeout
+                | ReconnError::MaxSlotsExceeded
+                | ReconnError::Call(_)
+                | ReconnError::ParseError(_)
+                | ReconnError::HttpNotImplemented
+                | ReconnError::EmptyBatchRequest(_)
+                | ReconnError::RegisterMethod(_) => unreachable!(),
+            }
+        }
+
+        async {
+            let sub = self
+                .subscribe_raw(sub.to_string(), params, unsub.to_string())
+                .await
+                .map_err(|e| RpcError::ClientError(Box::new(e)))?;
+
+            let id = match sub.id() {
+                SubscriptionId::Num(n) => n.to_string(),
+                SubscriptionId::Str(s) => s.to_string(),
+            };
+            let stream = sub
+                .map_err(|e| RpcError::DisconnectedWillReconnect(Some(rpc_error(e.0))))
```

Comment:
Can it ever be `None`?

Reply:
No, but it's for other implementations that may not be able to fetch the close reason.

Reply:
Good point; They _could_ just use an empty string or return "unknown reason" in that case, but happy with `None` too if you prefer that :)

Reply:
fine, String is probably nice anyway